src/main/java/clover/util/Storage.java

```diff
@@ -0,0 +1,88 @@
+package clover.util;
+
+import clover.task.Deadline;
+import clover.task.Event;
+import clover.task.Task;
+import clover.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public ArrayList<Task> load() throws IOException {
+        ArrayList<Task> allTasks = new ArrayList<>();
+        File file = new File(filePath);
+
+        if (!file.exists()) { //if file doesn't exist at the start
+            createFile(file);
+            return allTasks;
+        }
+
+        //else, file does exist
+        Scanner scanner = new Scanner(file);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+
+            /*
+            [0]: clover.task.Task, clover.task.Deadline or clover.task.Event
+            [1]: 1 or 0
+            [2]: task
+            [3]: due date
+            [4]: time
+             */
+            String[] split = line.split(" \\| "); //split up line at even "|"
+            Task task;
+
+            switch (split[0]) {
```

Comment:
Switch block has the wrong indenting, should be 

```
switch (split[0]) {
case
```

Reply:
Good catch!